src/rule-unused-fragments.js

```diff
@@ -0,0 +1,136 @@
+        return;
+      }
+      node.selectionSet.selections.forEach(selection => {
+        walkAST(selection);
+      });
+      return;
+    }
+    for (const prop in node) {
+      const value = node[prop];
+      if (prop === 'loc') {
+        continue;
+      }
+      if (value && typeof value === 'object') {
+        walkAST(value);
+      } else if (Array.isArray(value)) {
+        value.forEach(child => {
+          walkAST(child);
+        });
+      }
+    }
+  }
+
+  walkAST(graphQLAst);
+  return fragmentNames;
+}
+
+function getGraphQLFragmentDefinitionName(graphQLAst) {
+  const fragmentDefinition = graphQLAst.definitions.find(
+    definition => definition.kind === 'FragmentDefinition'
+  );
+  if (fragmentDefinition) {
+    return fragmentDefinition.name.value;
+  } else {
+    return null;
+  }
+}
+
+function isGraphQLTemplate(node) {
+  return (
+    node.tag.type === 'Identifier' &&
+    node.tag.name === 'graphql' &&
+    node.quasi.quasis.length === 1
+  );
+}
+
+function rule(context) {
+  let foundImportedModules = [];
+  let templateLiterals = [];
+
+  return {
+    Program(_node) {
+      foundImportedModules = [];
+      templateLiterals = [];
+    },
+    'Program:exit'(_node) {
+      const fragmentsInTheSameModule = [];
+      templateLiterals.forEach(templateLiteral => {
+        const graphQLAst = getGraphQLAST(templateLiteral);
+        if (!graphQLAst) {
+          // ignore nodes with syntax errors, they're handled by rule-graphql-syntax
+          return;
+        }
+        const fragmentName = getGraphQLFragmentDefinitionName(graphQLAst);
+        if (fragmentName) {
+          fragmentsInTheSameModule.push(fragmentName);
+        }
+      });
+
+      templateLiterals.forEach(templateLiteral => {
+        const graphQLAst = getGraphQLAST(templateLiteral);
+        if (!graphQLAst) {
+          // ignore nodes with syntax errors, they're handled by rule-graphql-syntax
+          return;
+        }
+
+        const queriedFragments = getGraphQLFragmentNames(graphQLAst);
+        for (const fragment in queriedFragments) {
+          const matchedModuleName = foundImportedModules.find(name =>
+            fragment.startsWith(name)
+          );
+          if (
+            !matchedModuleName &&
+            !fragmentsInTheSameModule.includes(fragment)
+          ) {
+            context.report({
+              node: templateLiteral,
+              loc: utils.getLoc(
+                context,
+                templateLiteral,
+                queriedFragments[fragment]
+              ),
+              message:
+                `This queries for the fragment \`${fragment}\` but this file does ` +
+                'not seem to use it directly. If a different file needs this ' +
+                'information that file should export a fragment and colocate ' +
+                'the query for the data with the usage.\n'
+            });
+          }
+        }
+      });
+    },
+    ImportDeclaration(node) {
```

Comment:
You'll also want to handle skipping `import type` (importing type definitions, supported by both TypeScript and Flow), and imports using `require`. For completeness, you'll also want to handle `ImportExpression`. Examples:

```js
import type {Component} from 'FancyTypes'; // you'll want to skip these, as they are types that get compiled away
const Component = require('Component.react');
const promise = import("module-name");
```

Also, to reduce the # of false positives, we also track the `Identifier` name for the `CallExpression`. This is to support scenarios where someone renames the imported module:

```js
const RenamedComponent = require('Component.react');
```

Reply:
Hi, not sure I understand what do you mean here. This rule track only imported module names but not the usage of imported values.

Reply:
I think you got it!